src/SourceLocator/Type/PhpInternalSourceLocator.php

```diff
@@ -16,9 +18,9 @@
      */
     private $stubber;
 
-    public function __construct()
+    public function __construct(?Locator $locator = null)
```

Comment:
IMO, the dependency should be the parser, and the dependency shouldn't be optional.